crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -815,29 +809,35 @@ where
     Ok(async move {
         let archiver = match maybe_archiver {
             Some(archiver) => archiver,
-            None => initialize_archiver(&segment_headers_store, &subspace_link, client.as_ref())?,
+            None => {
+                let Some(last_segment_header) = segment_headers_store.last_segment_header() else {
+                    unreachable!("Already checked above that max_segment_index is not None")
+                };
+                let last_archived_block = last_segment_header.last_archived_block();
+
+                // wait for unknown blocks to arrive.
+                while let Some(ref block_import_notification) =
+                    block_importing_notification_stream.next().await
+                {
+                    if block_import_notification
+                        .block_number
+                        .checked_sub(&(last_archived_block.number + 1).into())
+                        .is_none()
+                    {
+                        break;
+                    }
+                }
+                initialize_archiver(&segment_headers_store, &subspace_link, client.as_ref())?
+            }
         };
 
         let InitializedArchiver {
             confirmation_depth_k,
             mut archiver,
-            older_archived_segments,
             best_archived_block,
         } = archiver;
         let (mut best_archived_block_hash, mut best_archived_block_number) = best_archived_block;
 
-        let archived_segment_notification_sender =
-            subspace_link.archived_segment_notification_sender.clone();
-
-        // Farmers may have not received all previous segments, send them now.
-        for archived_segment in older_archived_segments {
```

Comment:
This was removed correctly, thanks! In the past farmer was relying on getting all notifications exactly in order, but that is not a requirement for a long time.